src/asami/entities.cljc

```diff
@@ -174,20 +174,32 @@
                                (if-let [overflow (:overflow (ex-data e))]
                                  (reduced last-result)
                                  (throw e))))
-                           (if (and (seqable? obj)
+                           (if (and (seqable? obj)          ; would it be ok performance-wise to turn this into if-let with destructuring to extract e a v to use below instead of (nth ...)?
```

Comment:
I would much prefer destructuring, but I opted for nth here because it allows us to extract fewer items depending on what is read, and this is very hot code.
The tradeoff here is performance vs clarity, but given how much data runs through it, then performance is being prioritized.

Reply:
I thought so. Thank you for the explanation!